go/vt/vtgate/planbuilder/operators/join.go

```diff
@@ -61,7 +63,7 @@ func (j *Join) SetInputs(ops []Operator) {
 }
 
 func (j *Join) Compact(ctx *plancontext.PlanningContext) (Operator, *ApplyResult) {
-	if j.LeftJoin {
+	if !j.JoinType.IsCommutative() {
 		// we can't merge outer joins into a single QG
 		return j, NoRewrite
```

Comment:
comment not aligning with the condition check

Reply:
2nd question should we not be allowed to merge keeping the original order for straight join